docs/eip-draft_bls_remote_signer_api.md

```diff
@@ -0,0 +1,418 @@
+* TCP_NODELAY set
+* Connected to localhost (127.0.0.1) port 9000 (#0)
+> POST /sign/b7354252aa5bce27ab9537fd0158515935f3c3861419e1b4b6c8219b5dbd15fcf907bddf275442f3e32f904f79807a2a HTTP/1.1
+> Host: localhost:9000
+> User-Agent: curl/7.68.0
+> Accept: */*
+> Content-Type: application/json
+> Content-Length: 84
+>
+* upload completely sent off: 84 out of 84 bytes
+* Mark bundle as not supporting multiuse
+< HTTP/1.1 500 Internal Server Error
+< content-length: 43
+< date: Wed, 30 Sep 2020 02:21:08 GMT
+<
+* Connection #0 to host localhost left intact
+{"error":"Storage error: PermissionDenied"}
+```
+
+## Implementation
+
+Repository Url | Language | Organization | Commentary
+--- | --- | --- | ---
+[BLS Remote Signer](https://github.com/sigp/rust-bls-remote-signer) | Rust | Sigma Prime | Supports proposed specification.
+[Web3signer](https://github.com/PegaSysEng/web3signer) | Java | PegaSys | Supports proposed specification, although with [slightly different methods](https://pegasyseng.github.io/web3signer/web3signer-eth2.html):<br>{`/sign` => `/api/v1/eth2/sign`, `/publicKeys` => `/api/v1/eth2/publicKeys`}.
+
+The Prysm client supports a [Remote Signing Wallet](https://docs.prylabs.network/docs/wallet/remote/), however its API requires using gRPC as transport.
+
+## Security Considerations
+
+### Threat model
+
+Let's consider the following threats and their mitigations:
+
+Threat | Mitigation(s)
+--- | ---
+An attacker can spoof the validator client. | See the discussion at [Authentication](#authentication).
+An attacker can send a crafted message to the signer. | See discussion at  [Pre-image validation](#pre-image-validation).
+An attacker can create, update, or delete secret keys. | Keys are not to be writable via any interface of the remote signer.
+An attacker can repudiate a sent message. | Implement logging in the signer. Enhance it by sending logs to a syslog box.
+An attacker can disclose the contents of a private key by retrieving the key from storage. | Storage in Hardware security module (HSM).<br>_or_<br>Storage in Secrets management applications (e.g. Hashicorp Vault).
+An attacker can eavesdrop on the uploading of a secret key. | Upload the keys using a secure channel, based on each storage specification.
+An attacker can eavesdrop on the retrieval of a key from the remote signer. | Always pass the data between storage and remote signer node using a secure channel.
+An attacker can dump the memory in the remote signer to disclose a secret key. |  Prevent physical access to the node running the remote signer.<br>_or_<br>Prevent access to the terminal of the node runnig the remote signer: Logs being sent to a syslog box. Deployments triggered by a simple, non-parameterized API.<br>_or_<br>Implement zeroization of the secret key at memory.<br>_or_<br>Explore the compilation and running of the remote signer in a Trusted execution environment (TEE).
+An attacker can DoS the remote signer. | Mitigation left to the implementer.
```

Comment:
Perhaps worth adding a few pointers to the DoS mitigation, such as:
- IP filtering
- Rate limiting

Reply:
:heavy_check_mark: